### PR TITLE
Use new krb5_principal and krb5_keytab LWRPs

### PR DESCRIPTION
**Pull request by wolf31o2**
- Replace all instances of krb5_utils with the LWRPs
- Create HTTP principal independent of others
- Add HTTP principal to service keytabs, only
- Update version dependencies on krb5 cookbook

### CODE REVIEW

File: Berksfile [hunk 1]
```diff
@@ -12,5 +12,4 @@ if RUBY_VERSION.to_f < 2.0
   cookbook 'ohai', '< 4.0'
 end
 
-cookbook 'hadoop', git: 'https://github.com/caskdata/hadoop_cookbook.git', branch: 'master'
```
Comment (dereklwood) [inline Berksfile]: what's the reason for this?  I assume this means we must test against the release version, and not whatever's merged to master, which means we can't release these cookbooks together.  not sure what a use-case for that would be though.

Comment (wolf31o2) [inline Berksfile]: It was added long ago, for that reason. We needed something that wasn't released. I'm fine with keeping this, but prefer to test using release materials, where possible.

Comment (wolf31o2) [inline Berksfile]: Added it back.

**Comment by dereklwood** [comment]
👍 lgtm

**Pull request merged**